chore(deps): update packages

QZera opened the pull request:
- Updated outdated packages to their latest version
- Fixed a typescript error in `components/TaskKindSelect.vue`

Closes [fdbcb15](https://app.radicle.xyz/nodes/seed.radicle.garden/rad:z2BdUVZFvHdxRfdtGJQdSH2kyXNM6/issues/fdbcb1542abb4fd56381f3dd2f453efb290104a6)

maninak commented on package.json:
```diff
@@ -21,26 +21,26 @@
-    "typescript": "^5.4.2",
-    "vue": "^3.4.21",
-    "vue-router": "^4.3.0"
+    "nuxt": "^3.11.2",
```
Is there something keeping us from pushing it to "3.12.x"?

maninak commented on components/TaskKindSelect.vue:
```diff
@@ -50,7 +50,7 @@ function handleChange(selected: (typeof options)[number]) {
       id="task-kind-select"
       :model-value="selected"
-      :options="options"
+      :options="options as unknown as string[]"
```
Here the need for the type assertions come from the [`as const`](https://github.com/cytechmobile/radicle-planning-boards/pull/75/files#diff-14fcf142c2c0d64d7e80e9764f8eab4c43f16c02db2bf775381ae09951fa3a06R4) on line 4 of the same file. The `as const` makes sense to keep as is but we could make the type assertion cleaner.

Since the `as const` modifies the array to be `readonly ["All", "Issues", "Patches"]` it no longer extends `string[]` because of the `readonly` part. A cleaner approach would be to define a new helper type `type Writeable<T> = { -readonly [P in keyof T]: T[P] }` (save it as non-exported in the top of this file or under `/types/helpers.ts`.

Here's how it all looks in the end:

![image](https://github.com/cytechmobile/radicle-planning-boards/assets/7542054/10eaea6b-41c4-430d-a23b-11e28138ea17)


The above should also resolve this https://github.com/cytechmobile/radicle-planning-boards/pull/73/files#file-components-taskkindselect-vue-L53 on another PR

![image](https://github.com/cytechmobile/radicle-planning-boards/assets/7542054/1f7166ec-cfe2-4d5d-b57b-388a9acd3c93)

maninak reviewed (changes requested):
I haven't gone through the changelog of each of those deps to make sure there isn't some new feature we could take advantage of, I'm leaving that to you. You should apply any change in this PR if it's a small one per dep or create a new ticket if it's bigger change.

Other than that let's ensure we're switching _at least for nuxt_ (and deps that require it) on its latest version.

Also that typing issue we can handle better.

maninak commented on pnpm-lock.yaml:
This should also read `lockfileVersion: '9.0'`. Make sure you're using the latest version of pnpm, there's important changes they've released in the latest ones and we should be aligned there too.

QZera replied:
> Done!

Turns out lockfile v9 is causing the Netlify build to fail. I've had similar issues with Vercel which is the reason I'm still using pnpm 8

maninak commented:
Related to the [previous comment](https://github.com/cytechmobile/radicle-planning-boards/pull/75#discussion_r1636333900), on line 14 of `package.json` (https://github.com/cytechmobile/radicle-planning-boards/pull/75/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R14) we should change this to be

```diff
-    "verify-deps": "npx pnpm i --frozen-lockfile --prefer-offline",
+    "verify-deps": "npx --yes pnpm@latest i --frozen-lockfile --prefer-offline",
```

This would also use the latest version of pnpm, not a possibly outdated locally cached one.

QZera commented:
> Related to the [previous comment](https://github.com/cytechmobile/radicle-planning-boards/pull/75#discussion_r1636333900), on line 14 of `package.json` (https://github.com/cytechmobile/radicle-planning-boards/pull/75/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R14) we should change this to be
> 
> ```diff
> -    "verify-deps": "npx pnpm i --frozen-lockfile --prefer-offline",
> +    "verify-deps": "npx --yes pnpm@latest i --frozen-lockfile --prefer-offline",
> ```
> 
> This would also use the latest version of pnpm, not a possibly outdated locally cached one.

I don't think that's a good idea. As mentioned in [this thread](https://github.com/cytechmobile/radicle-planning-boards/pull/75#discussion_r1641740706) pnpm 9 is causing issues with Netlify. And in my personal experience, hosting services aren't guaranteed to support that latest version of package managers from day 1.

maninak commented:
> > Related to the [previous comment](https://github.com/cytechmobile/radicle-planning-boards/pull/75#discussion_r1636333900), on line 14 of `package.json` ([#75 (files)](https://github.com/cytechmobile/radicle-planning-boards/pull/75/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R14)) we should change this to be
> > ```diff
> > -    "verify-deps": "npx pnpm i --frozen-lockfile --prefer-offline",
> > +    "verify-deps": "npx --yes pnpm@latest i --frozen-lockfile --prefer-offline",
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This would also use the latest version of pnpm, not a possibly outdated locally cached one.
> 
> I don't think that's a good idea. As mentioned in [this thread](https://github.com/cytechmobile/radicle-planning-boards/pull/75#discussion_r1641740706) pnpm 9 is causing issues with Netlify. And in my personal experience, hosting services aren't guaranteed to support that latest version of package managers from day 1.

If we don't want to go with `pnpm@latest` then we should still use the suggested changes (notice also the new `--yes`) and lock it at the version we are supporting _and expecting_ e.g. pnpm@^8.

For you with your machinesetup, it may seem like it works without it, but for anyone else cloning the repo without those changes, they might have a bumpy first ride.

QZera commented on nuxt.config.ts:
```diff
@@ -4,6 +4,9 @@ const defaultParentOrigin = 'https://guileless-puppy-2d91b5.netlify.app'
 // https://nuxt.com/docs/api/configuration/nuxt-config
 export default defineNuxtConfig({
+  future: {
+    compatibilityVersion: 4,
```
> This should not have any effect unless our nuxt is also updated to the nightly build

This was actually added to all builds with [v3.12.0](https://nuxt.com/blog/v3-12#testing-nuxt-4-changes)

--- 

Reverted the change.

QZera commented:
> > > Related to the [previous comment](https://github.com/cytechmobile/radicle-planning-boards/pull/75#discussion_r1636333900), on line 14 of `package.json` ([#75 (files)](https://github.com/cytechmobile/radicle-planning-boards/pull/75/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R14)) we should change this to be
> > > ```diff
> > > -    "verify-deps": "npx pnpm i --frozen-lockfile --prefer-offline",
> > > +    "verify-deps": "npx --yes pnpm@latest i --frozen-lockfile --prefer-offline",
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > This would also use the latest version of pnpm, not a possibly outdated locally cached one.
> > 
> > 
> > I don't think that's a good idea. As mentioned in [this thread](https://github.com/cytechmobile/radicle-planning-boards/pull/75#discussion_r1641740706) pnpm 9 is causing issues with Netlify. And in my personal experience, hosting services aren't guaranteed to support that latest version of package managers from day 1.
> 
> If we don't want to go with `pnpm@latest` then we should still use the suggested changes (notice also the new `--yes`) and lock it at the version we are supporting _and expecting_ e.g. `pnpm@^8`. We could also add this as [a requirement in package.json](https://stackoverflow.com/questions/71747609/how-to-specify-packagemanager-in-package-json).
> 
> For you with your machine setup, it may seem like it works without it, but for anyone else cloning the repo without those changes, they might have a bumpy first ride.

Agreed. As I was doing this I came across another issue with Netlify, it doesn't support the full spec for the `packageManager` option. In our case this means we can't use `~` or `^` and need to use a specific pnpm version.

While looking into that, I realised we *could* actually use pnpm 9 by specifying it in the `packageManager` option. So this is what I have done.

---

TLDR; 

- Updated the script with `pnpm@latest-9`
- Added `packageManager` option with `pnpm@9.3.0` (latest as of time of writing)

maninak commented:
Yay, teamwork! This is a solid chore patch, well done! :clap:

maninak reviewed (approved):
Yay, teamwork! This is a solid chore patch, well done! :clap:

QZera commented:
@maninak Need re-review after resolving some conflicts. Also removed the caret from `@nuxt/ui` version.

maninak commented:
> Also removed the caret from `@nuxt/ui` version.

Care to elaborate on the reason why? Sounds important.

QZera commented:
> > Also removed the caret from `@nuxt/ui` version.
> 
> Care to elaborate on the reason why? Sounds important.

Updating to v2.16.x onwards is causing an out of memory error as mentioned in [this thread](https://github.com/cytechmobile/radicle-planning-boards/pull/75#discussion_r1641241742)
